### PR TITLE
fix: builtin swc loader input sourcemap

### LINKED ISSUE #5480: builtin:swc-loader breaks source-map-loader
### System Info

```
  System:
    OS: macOS 13.6.4
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 16.17 MB / 16.00 GB
    Shell: 3.7.0 - /usr/local/bin/fish
  Binaries:
    Node: 16.17.0 - ~/.local/share/mise/installs/node/16.17.0/bin/node
    npm: 8.15.0 - ~/.local/share/mise/installs/node/16.17.0/bin/npm
  Browsers:
    Chrome: 121.0.6167.85
    Edge: 120.0.2210.144
    Firefox: 122.0
    Safari: 17.3
  npmPackages:
    @rspack/cli: ^0.5.2 => 0.5.2 
    @rspack/core: ^0.5.2 => 0.5.2 
    @rspack/plugin-react-refresh: ^0.4.5 => 0.4.5 
```

### Details

Basically what this says on the tin. It seems like the builtin swc-loader is not using `inputSourceMap` properly. This prevents any sourcemaps for libraries that are loaded with `source-map-loader` from being merged into the final output sourcemap, and instead you mostly end up with references to bundled or minified source code in the sourcemap.

The normal webpack `swc-loader` implementation does not have this issue. As a workaround for now, I've just reverted back to using that.

### Reproduce link

_No response_

### Reproduce Steps

```js
{
  module: {
    rules: [
      {
        test: /\.[tj]sx?$/,
        loader: 'builtin:swc-loader',
        options: {
          parser: {
            syntax: 'typescript',
            tsx: true,
          },
        },
      },
      {
        test: /\.[tj]sx?$/,
        enforce: 'pre',
        loader: 'source-map-loader',
      },
    ],
  },
}
```

### PR DESCRIPTION
**Pull request by 2239559319**
## Summary

fix #5480

`std::mem::take` will be replaced with the default value. Therefore, subsequent loaders cannot obtain the correct value.

## Require Documentation?

- [x] No
- [ ] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

**Comment by ahabhgk** [comment]
Could you add a test?

**Comment by 2239559319** [comment]
> Could you add a test?

test has been added and passed in my repository see [https://github.com/2239559319/rspack/pull/3](repository) but failed in this repository.

**Comment by hardfist** [comment]
!bench

File: webpack-test/configCases/source-map/srouce-map-loader/index.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+it("should include a.ts in SourceMap", function() {
```
Comment (h-a-n-a) [inline webpack-test/configCases/source-map/srouce-map-loader/index.js:1]: Additional test cases should be added to `packages/rspack/tests`. Would you please move?

Comment (2239559319) [reply webpack-test/configCases/source-map/srouce-map-loader/index.js]: Folder moved.

File: packages/rspack/tests/configCases/source-map/srouce-map-loader/index.js
Comment (VenDream) [inline packages/rspack/tests/configCases/source-map/srouce-map-loader/index.js:1]: Maybe there is a typo on the folder name.
`srouce` -> `source`

Comment (2239559319) [reply packages/rspack/tests/configCases/source-map/srouce-map-loader/index.js:1]: moved, cr please

**Comment by sgarfinkel** [comment]
@ahabhgk Can we get a review on this please?

**Review by ahabhgk — APPROVED** [review]
Sorry for delay, this is a good catch, thank you!

**Comment by littleprincewdk** [comment]
@hardfist Is this released？the latest version is still not effective

compile this file，sources should be **.ts**(webpack `swc-loader` is ok)，but `builtin:swc-loader` will get **.js**，prove that it does not take effect
```javascript
"use strict";

function message() {
  console.log('inline-source-map');
}

message();
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImlubGluZS1zb3VyY2UtbWFwLnRzIl0sIm5hbWVzIjpbIm1lc3NhZ2UiLCJjb25zb2xlIiwibG9nIl0sIm1hcHBpbmdzIjoiOztBQUFBLFNBQVNBLE9BQVQsR0FBbUI7QUFDakJDLEVBQUFBLE9BQU8sQ0FBQ0MsR0FBUixDQUFZLG1CQUFaO0FBQ0Q7O0FBRURGLE9BQU8iLCJzb3VyY2VzQ29udGVudCI6WyJmdW5jdGlvbiBtZXNzYWdlKCkge1xuICBjb25zb2xlLmxvZygnaW5saW5lLXNvdXJjZS1tYXAnKTtcbn1cblxubWVzc2FnZSgpO1xuIl19
```

**Comment by ckken** [comment]
> @hardfist Is this released？the latest version(0.6.5) is still not effective  compile this file，sources should be **.ts**(webpack `swc-loader` is ok)，but `builtin:swc-loader` will get **.js**，prove that it does not take effect
> 
> ```js
> "use strict";
> 
> function message() {
>   console.log('inline-source-map');
> }
> 
> message();
> //# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImlubGluZS1zb3VyY2UtbWFwLnRzIl0sIm5hbWVzIjpbIm1lc3NhZ2UiLCJjb25zb2xlIiwibG9nIl0sIm1hcHBpbmdzIjoiOztBQUFBLFNBQVNBLE9BQVQsR0FBbUI7QUFDakJDLEVBQUFBLE9BQU8sQ0FBQ0MsR0FBUixDQUFZLG1CQUFaO0FBQ0Q7O0FBRURGLE9BQU8iLCJzb3VyY2VzQ29udGVudCI6WyJmdW5jdGlvbiBtZXNzYWdlKCkge1xuICBjb25zb2xlLmxvZygnaW5saW5lLXNvdXJjZS1tYXAnKTtcbn1cblxubWVzc2FnZSgpO1xuIl19
> ```
> 
> 
> repo: https://github.com/littleprincewdk/rspack-inputSourceMap-bug-demo

same here

**Comment by h-a-n-a** [comment]
> @hardfist Is this released？the latest version(0.6.5) is still not effective  compile this file，sources should be **.ts**(webpack `swc-loader` is ok)，but `builtin:swc-loader` will get **.js**，prove that it does not take effect
> 
> ```js
> "use strict";
> 
> function message() {
>   console.log('inline-source-map');
> }
> 
> message();
> //# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImlubGluZS1zb3VyY2UtbWFwLnRzIl0sIm5hbWVzIjpbIm1lc3NhZ2UiLCJjb25zb2xlIiwibG9nIl0sIm1hcHBpbmdzIjoiOztBQUFBLFNBQVNBLE9BQVQsR0FBbUI7QUFDakJDLEVBQUFBLE9BQU8sQ0FBQ0MsR0FBUixDQUFZLG1CQUFaO0FBQ0Q7O0FBRURGLE9BQU8iLCJzb3VyY2VzQ29udGVudCI6WyJmdW5jdGlvbiBtZXNzYWdlKCkge1xuICBjb25zb2xlLmxvZygnaW5saW5lLXNvdXJjZS1tYXAnKTtcbn1cblxubWVzc2FnZSgpO1xuIl19
> ```
> 
> 
> repo: https://github.com/littleprincewdk/rspack-inputSourceMap-bug-demo

Would you please try the latest version? We fixed a case where there's a source map passing over from the previous loader to use it as the `inputSourceMap` for `builtin:swc-loader`. cc @littleprincewdk @ckken

**Comment by ckken** [comment]
@h-a-n-a rspack 0.7.0 no work also!

**Comment by ckken** [comment]
fixed it in rspack 0.7.0! nice

**Pull request merged**